stratify/_bounded_vinterp.py

```diff
@@ -71,41 +71,45 @@ def interpolate_conservative(z_target, z_src, fz_src, axis=-1):
 
     if src_shape != tgt_shape:
         src_shape = list(z_src.shape)
-        src_shape[axis_relative] = '-'
+        src_shape[axis_relative] = "-"
         tgt_shape = list(z_target.shape)
-        src_shape[axis_relative] = '-'
-        msg = ('Expecting the shape of the source and target levels except '
-               'the axis of interpolation to be identical.  {} != {}')
+        src_shape[axis_relative] = "-"
+        msg = (
+            "Expecting the shape of the source and target levels except "
+            "the axis of interpolation to be identical.  {} != {}"
+        )
         raise ValueError(msg.format(tuple(src_shape), tuple(tgt_shape)))
 
     dat_shape = list(fz_src.shape)
-    dat_shape = dat_shape[-(z_src.ndim-1):]
+    dat_shape = dat_shape[-(z_src.ndim - 1) :]
     src_shape = list(z_src.shape[:-1])
     if dat_shape != src_shape:
         dat_shape = list(fz_src.shape)
-        dat_shape[:-(z_src.ndim-1)] = '-'
-        msg = ('The provided data is not of compatible shape with the '
-               'provided source bounds. {} != {}')
+        dat_shape[: -(z_src.ndim - 1)] = "-"
+        msg = (
+            "The provided data is not of compatible shape with the "
+            "provided source bounds. {} != {}"
+        )
         raise ValueError(msg.format(tuple(dat_shape), tuple(src_shape)))
 
     if z_src.shape[-1] != 2:
-        msg = 'Unexpected source and target bounds shape. shape[-1] != 2'
+        msg = "Unexpected source and target bounds shape. shape[-1] != 2"
         raise ValueError(msg)
 
     # Define our source in a consistent way.
     # [broadcasting_dims, axis_interpolation, z_varying]
 
     # src_data
-    bdims = list(range(fz_src.ndim - (z_src.ndim-1)))
-    data_vdims = [ind for ind in range(fz_src.ndim) if ind not in
-                  (bdims + [axis])]
+    bdims = list(range(fz_src.ndim - (z_src.ndim - 1)))
+    data_vdims = [ind for ind in range(fz_src.ndim) if ind not in (bdims + [axis])]
     data_transpose = bdims + [axis] + data_vdims
     fz_src_reshaped = np.transpose(fz_src, data_transpose)
     fz_src_orig = list(fz_src_reshaped.shape)
     shape = (
-        int(np.product(fz_src_reshaped.shape[:len(bdims)])),
+        int(np.product(fz_src_reshaped.shape[: len(bdims)])),
         fz_src_reshaped.shape[len(bdims)],
-        int(np.product(fz_src_reshaped.shape[len(bdims)+1:])))
+        int(np.product(fz_src_reshaped.shape[len(bdims) + 1 :])),
```

Comment:
I would prefer `[(len(bdims) + 1):]`, adding spaces around slice op is not idiomatic

Reply:
All these changes to the actual code are a result of black auto-fornatting. I trying not to touch the code, other than  py2 migration

Reply:
But yeah... what can i say :scream: